stand/lua/config.lua

```diff
@@ -578,8 +578,35 @@ function config.readConf(file, loaded_files)
 
 	loaded_files[file] = true
 
+	-- If product_vars is set, it must be a list of environment variable names
+	-- to walk through to guess product information. The order matters as
+	-- reading a config file override the previously defined values.
+	--
+	-- And if product information can be guessed, for each product information
+	-- found, append:
+	-- - /boot/PRODUCT.conf to loader_conf_files
+	-- - and /boot/loader.conf.d/PRODUCT/ to loader_conf_dirs
+	local product_vars = getEnv("product_vars")
+	local products = {}
+	if product_vars ~= nil then
+		for var in product_vars:gmatch("%S+") do
+			local product = getEnv(var)
+			if product ~= nil then
+				table.insert(products, product)
+			end
+		end
+	end
+
 	-- Going to process "loader_conf_files" extra-files
 	local loader_conf_files = getEnv("loader_conf_files")
+	if products ~= nil then
+		local product_loader_conf_files = "/boot/" .. table.concat(products, ".conf /boot/") .. ".conf"
+		if loader_conf_files ~= nil then
+			loader_conf_files = loader_conf_files .. " " .. product_loader_conf_files
+		else
+			loader_conf_files = product_loader_conf_files
+		end
```

Comment:
Maybe we should append both the .conf and .lua versions of the file. We'll silently ignore the one that doesn't exist. We don't need to do the existence check that I thought we might because config.readConf will silently ignore missing files.